python/glow/wgr/linear_model/functions.py

```diff
@@ -499,18 +509,49 @@ def __check_binary(df: pd.DataFrame) -> None:
         df : Pandas DataFrame
     """
     for label in df:
-        unique_vals = [v for v in np.sort(df[label].unique()).tolist() if not np.isnan(v)]
-        if unique_vals != [0.0, 1.0]:
-            warnings.warn(f"Column {label} is not binary (unique vals found: {unique_vals})",
+        num_non_binary_vals = __num_non_binary_values(df[label])
+        if num_non_binary_vals != 0:
+            warnings.warn(f"Column {label} is not binary ({num_non_binary_vals} non binary values)",
                           UserWarning)
 
 
 @typechecked
-def validate_inputs(labeldf: pd.DataFrame, covdf: pd.DataFrame, label_type='continuous') -> None:
+def __check_binary_or_standardized(df: pd.DataFrame) -> None:
+    """
+    Warns if columns of a pandas DataFrame are neither binary nor standardized.
+
+    Args:
+        df : Pandas DataFrame
+    """
+    for label in df:
+        mean = df[label].mean()
+        std_dev = df[label].std()
+        num_non_binary_values = __num_non_binary_values(df[label])
+        continuous_ok = math.isclose(mean, 0, abs_tol=0.01) and math.isclose(
```

Comment:
I originally tried doing this, but it's awkward because we have error messages that refer to specific violations, so a boolean is not sufficient. I consolidated a few more things.